idaes/core/solvers/petsc.py

```diff
@@ -477,6 +477,13 @@ def petsc_dae_by_time_element(
                 variables,
                 deriv_diff_map,
             )
+            # We need to check if there are derivatives in the problem before
+            # sending this to the solver.  We'll assume that if you are using
+            # this and don't have any differential equations, you are making a
+            # mistake.
+            if len(differential_vars) < 1:
+                raise RuntimeError(
+                    "No differential equations found, you do not need a DAE solver.")
```

Comment:
This sounds good, but should this check happen earlier? E.g. at the end of `_get_derivative_differential_data_map`?

Reply:
Wait maybe it's not that hard.  I should be able to check that there is at least one derivative for each time point.  I think I can improve this.

Reply:
What happens if you send a model with no derivatives to Petsc? Does it just solve the square system?

Reply:
With your comment in mind, I'm fine with the check where it is, but I'll leave it up to you.

Reply:
Okay changed my mind again.  It's probably not worth the trouble of checking this earlier.

Reply:
> What happens if you send a model with no derivatives to Petsc? Does it just solve the square system?

It should but it crashes.  Most likely I messed up and assumed I had at least one derivative.  I’ll sort that out in the solver part eventually.  For now this check will stop it from getting there.